athena-views-pipeline/README.md

```diff
@@ -0,0 +1,148 @@
+```
+$ pip install aws-ddk
+```
+
+To verify the installation, run:
+
+```
+$ ddk --help
+```
+
+Create and activate a virtualenv
+
+```
+$ python -m venv .venv && source .venv/bin/activate
+```
+
+Install the dependencies from requirements.txt
+This is when the AWS DDK Core library is installed
+
+```
+$ pip install -r requirements.txt --no-cache-dir
+```
+
+If your default AWS region is not set then
+
+```
+export AWS_DEFAULT_REGION='<aws-region>'
+```
+
+# DDK Bootstrapping
+
+In order to deploy DDK apps, you need to bootstrap your environment with the correct environment name.
+
+Run the following command to bootstrap the `dev` environment for your respective AWS Account and Region:
+
+```
+$ ddk bootstrap --profile [AWS_PROFILE] --region [AWS_REGION]
+```
+
+# Edit DDK.json
+
+Open the `ddk.json` file in the top-level directory of this repo. 
+
+Edit the configuration file and enter your desired **AWS account number** and **AWS region** for the infrastructure to be deployed in.
+
+Be sure to save your changes to the file before proceeding!
+
+# How to define a view?
+
+This pattern contains folders corresponding to databases in glue catalog. In each of these folders, there is one file corresponding to each view available or to be created in glue catalog.
+
+For example, the query which underlies "person_sb"."view_person_gender_male" is located in `athena_views_pipeline/sql/person_db/view_person_gender_male.sql`.
+
+1. Create a new SQL file whose file name corresponds to the view name
+File must be put under the folder which corresponds to the database in the glue catalog where the view will be housed. The filename must be prefixed with view_.
+
+2. Write SQL query
+There is no need to write the CREATE OR REPLACE VIEW statement. This will be added automatically, only provide the actual query for the view.
+
+3. you can provide custom cron schedule by adding ```-- SCHEDULE: "cron(*/10 * * * ? *)"``` at the first line of the sql file, if it is not added a default schedule of ```-- SCHEDULE: "cron(*/5 * * * ? *)"``` will be created for that query.
+
+# Deploy the Data Pipeline
+
+To deploy the pipeline, run the following command:
+
+```
+$ ddk deploy --profile [AWS_PROFILE]
+```
+
+This command should launch a CloudFormation template in the AWS account and region specified in DDK.json, and should take a few minutes to create.
+
+Once the CloudFormation stack has been successfully created, your AWS account now has the data pipeline outlined in the architecture section of this readme! 
+
+# Lakeformation 
+
+If you have enabled lakeformation then you will need to add permissions manually for the roles, in this example, it relies on IAM.
+# Testing the Data Pipeline
+
+To test the data pipeline, you will first have to do some pre prep as this example does not provide a pipeline for ingestion and cataloging, it believes such functionality exists within your data lake architecture. It expects the DBs for the views you want to create to already exist
```

Comment:
Can we ship it with with a db in IaC?

Reply:
Creation of db/cataloging to the db is not part of this pipeline, the customer can have x number of dbs already where they want to use this example to create/replace views, the example assumes the use of IAM roles instead of Lakeformation, I wanted to call out so that when error occurs, the user knows that it might be due to lakeformation.

Reply:
Makes sense, thanks!